core/templates/pages/facilitator-dashboard-page/facilitator-dashboard-page-root.component.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Facilitator Dashboard page root component.
+ */
+
+import {Component, OnInit} from '@angular/core';
+import {Router} from '@angular/router';
+import {AppConstants} from 'app.constants';
+import {MetaTagData} from 'pages/base-root.component';
+import {AccessValidationBackendApiService} from 'pages/oppia-root/routing/access-validation-backend-api.service';
+import {PageHeadService} from 'services/page-head.service';
+
+@Component({
+  selector: 'oppia-facilitator-dashboard-page-root',
+  templateUrl: './facilitator-dashboard-page-root.component.html',
+})
+export class FacilitatorDashboardPageRootComponent implements OnInit {
+  title: string =
+    AppConstants.PAGES_REGISTERED_WITH_FRONTEND.FACILITATOR_DASHBOARD.TITLE;
+
+  meta: MetaTagData[] = AppConstants.PAGES_REGISTERED_WITH_FRONTEND
+    .FACILITATOR_DASHBOARD.META as unknown as Readonly<MetaTagData>[];
+
+  pageIsShown: boolean = false;
+
+  constructor(
+    private avbas: AccessValidationBackendApiService,
+    private pageHeadService: PageHeadService,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.avbas.validateAccessToFacilitatorDashboardPage().then(
+      () => {
+        this.pageIsShown = true;
+        this.pageHeadService.updateTitleAndMetaTags(
+          AppConstants.PAGES_REGISTERED_WITH_FRONTEND.FACILITATOR_DASHBOARD
+            .TITLE,
+          AppConstants.PAGES_REGISTERED_WITH_FRONTEND.FACILITATOR_DASHBOARD.META
+        );
+      },
+      () => {
+        this.router.navigate([
+          `${AppConstants.PAGES_REGISTERED_WITH_FRONTEND.ERROR.ROUTE}/404`,
+        ]);
```

Comment:
Please use routeGaurd to 404 pages. (Follow the pattern/PR suggested in the #19435 )

Reply:
Done